tools/rosgraph/src/rosgraph/xmlrpc.py

```diff
@@ -122,38 +129,43 @@ def insert_multicall_client_ip( self, doc ):
                     else:
                         self.logger.warn( "multicall: unexpected entry %s" % name.data )
                 if methodName in self.secure_methods:
-                    ddoc = parseString( "<value><string>%s</string></value>\n" % self.client_ip )
-                    d = ddoc.firstChild.cloneNode(True)
-                    datas.appendChild(d)
+                    if methodName == "shutdown":
```

Comment:
Maybe a comment:
# Shutdown has an optional argument 'msg'. To ensure the correct dispatch is done we fill it here if not provided.